Subject: [PATCH] ohci1394: cleanup the "Unexpected PCI resource length" warning.
From: Jody McIntyre <scjody@modernduck.com>
Date: 1139973735 -0500

This warning happens in practice because the resource length reported by
the chipset is too large.  This is not actually a problem, so don't warn
about it.  If it happens to be too small, warn about that, but with
a different message so people who are used to ignoring the old message
don't.

Signed-off-by: Jody McIntyre <scjody@modernduck.com>
Signed-off-by: Olaf Hering <olh@suse.de>

---

 drivers/ieee1394/ohci1394.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

e4b38567a0af12016450003f7f7550d1b24732b3
diff --git a/drivers/ieee1394/ohci1394.c b/drivers/ieee1394/ohci1394.c
index 24d045c..ab01a54 100644
--- a/drivers/ieee1394/ohci1394.c
+++ b/drivers/ieee1394/ohci1394.c
@@ -3262,8 +3262,8 @@ static int __devinit ohci1394_pci_probe(
 	 * fail to report the right length.  Anyway, the ohci spec
 	 * clearly says it's 2kb, so this shouldn't be a problem. */
 	ohci_base = pci_resource_start(dev, 0);
-	if (pci_resource_len(dev, 0) != OHCI1394_REGISTER_SIZE)
-		PRINT(KERN_WARNING, "Unexpected PCI resource length of %lx!",
+	if (pci_resource_len(dev, 0) < OHCI1394_REGISTER_SIZE)
+		PRINT(KERN_WARNING, "PCI resource length of 0x%lx too small!",
 		      pci_resource_len(dev, 0));
 
 	/* Seems PCMCIA handles this internally. Not sure why. Seems
-- 
1.0.6
